### ISSUE TITLE
About training, validation and testing data

### ISSUE THREAD

**Issue by jieyongshi** [gabrieltseng/solar-panel-segmentation#6]
Good work! Could you please share more information about training, validation and testing data in detail? Such as how many images they have.

**Comment by gabrieltseng** [comment]
Hi!

Unfortunately, I no longer have the data so I can't check exactly, but:

An image is created centered on the centroid of each polygon. Since the [dataset](https://www.nature.com/articles/sdata2016106) describes over 19,000 images, I  would expect the train / val / test to be:

Train images: 80% = > 15,200 images
Val images: 10% = > 1,900 images
Test images: 10% = > 1,900 images

**Comment by jieyongshi** [comment]
Hi!
Thanks for your reply. You got good results. I use UNet to identify solar panels, but there are some false negatives, and some segmentation results are coarse. May I ask if you have any good advice to solve these problems?

**Comment by gabrieltseng** [comment]
Hmm, it's hard to say without seeing the code - what is the resolution of the input satellite data?

In addition, is it on the same dataset? If not, it might be worth using this dataset to supplement yours.

**Comment by jieyongshi** [comment]
> Hmm, it's hard to say without seeing the code - what is the resolution of the input satellite data?
> 
> In addition, is it on the same dataset? If not, it might be worth using this dataset to supplement yours.

I use the same data with you. The resolution of the data is 0.3m.

**Comment by gabrieltseng** [comment]
Ah, I see. In this case, here are two things which might help:

1. Images are generated per panel, so there is likely to be overlap. This might overestimate performance in the test set (it also serves as an image augmentation technique, since images will be at lots of different crops)
2. The U-Net's backbone is pre-trained to classify whether or not panels are in the image or not - this might help too.

**Comment by jieyongshi** [comment]
> Ah, I see. In this case, here are two things which might help:
> 
> 1. Images are generated per panel, so there is likely to be overlap. This might overestimate performance in the test set (it also serves as an image augmentation technique, since images will be at lots of different crops)
> 2. The U-Net's backbone is pre-trained to classify whether or not panels are in the image or not - this might help too.

Thanks a lot! Haven't you been working on the solar panel segmentation lately?

**Comment by gabrieltseng** [comment]
> Thanks a lot! Haven't you been working on the solar panel segmentation lately?

No - this was my (first and so far) last experiment with it
